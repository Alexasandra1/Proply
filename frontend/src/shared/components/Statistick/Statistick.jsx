import React from 'react';
import { TextBlock } from '../TextBlock/TextBlock';
import './Statistick.scss';

export function Statistick(props){
  return (
    <div className="statistick">
      {Array(4).fill().map((_, i) => (
        <div className={`statistick-block statistick-block-${i}`} key={i}>
          <TextBlock 
            styleTextBlock="text-block_statistick"
            styleTitle="title_statistick" 
            styleText="text_statistick" 
            title={props.titles[i]} 
            text={props.texts[i]} 
          />
        </div>
      ))}
    </div>
  );
}
