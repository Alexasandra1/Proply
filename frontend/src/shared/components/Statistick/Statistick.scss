.statistick {
  display: flex;
  width: 100%;
  justify-content: space-between;
  flex-wrap: wrap;
}

.statistick-block {
  width: calc(25% - 1px);
  height: 200px;
  border-right: 1px solid;
  display: flex;
  justify-content: center;
  align-items: center;
}

@media (max-width: 740px) {
  .statistick-block {
    width: calc(50% - 1px);
  }
}

@media (max-width: 340px) {
  .statistick-block {
    width: calc(100% - 1px);
  }
}
