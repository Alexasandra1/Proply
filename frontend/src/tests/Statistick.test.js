import React from 'react';
import renderer from 'react-test-renderer';
import { Statistick } from '../shared/components/Statistick/Statistick';

test('renders correctly', () => {
  const titles = ['Title 1', 'Title 2', 'Title 3', 'Title 4'];
  const texts = ['Text 1', 'Text 2', 'Text 3', 'Text 4'];
  const tree = renderer
    .create(<Statistick titles={titles} texts={texts} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
