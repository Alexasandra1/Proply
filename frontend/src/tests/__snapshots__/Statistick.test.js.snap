// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="statistick"
>
  <div
    className="statistick-block statistick-block-0"
  >
    <div
      className="text-block text-block_statistick"
    >
      <h1
        className="title title_statistick"
      >
        Title 1
      </h1>
      <p
        className="text text_statistick"
      >
        Text 1
      </p>
    </div>
  </div>
  <div
    className="statistick-block statistick-block-1"
  >
    <div
      className="text-block text-block_statistick"
    >
      <h1
        className="title title_statistick"
      >
        Title 2
      </h1>
      <p
        className="text text_statistick"
      >
        Text 2
      </p>
    </div>
  </div>
  <div
    className="statistick-block statistick-block-2"
  >
    <div
      className="text-block text-block_statistick"
    >
      <h1
        className="title title_statistick"
      >
        Title 3
      </h1>
      <p
        className="text text_statistick"
      >
        Text 3
      </p>
    </div>
  </div>
  <div
    className="statistick-block statistick-block-3"
  >
    <div
      className="text-block text-block_statistick"
    >
      <h1
        className="title title_statistick"
      >
        Title 4
      </h1>
      <p
        className="text text_statistick"
      >
        Text 4
      </p>
    </div>
  </div>
</div>
`;
